lib/TPP/Transforms/VectorContractToAMX.cpp

```diff
@@ -36,6 +36,113 @@ using namespace mlir;
 using namespace mlir::tpp;
 
 namespace {
+// Helper function to down convert and copy the result back to the output
+// matrix.
+static void downConvertAndCopyResult(OpBuilder &rewriter, Location loc,
+                                     Value accBuffer, Value accSubview,
+                                     MemRefType bufferType, ShapedType accType,
+                                     Value c0, Value mBound, Value nBound,
+                                     Value one, Value sixteen) {
```

Comment:
Could you generalize these parameters a bit?
I guess `accBuffer` is more of source and `accSubview` destination?
If you need arith constants just create them directly within this function.